### ISSUE TITLE
Progress bar for long text

### ISSUE THREAD

**Issue by fakerybakery** [yl4579/StyleTTS2#54]
Hi,
Might it be possible to implement a `tqdm` progress bar for longer text?
Thanks!

**Comment by yl4579** [comment]
I think so far it’s not very good at narrating the entire audiobook because the training data isn’t the entire audiobook. The training data is purely independent clip taken from audiobooks, rather than an entire audiobook. It won’t be like ElevenLabs that are trained with professional audiobook clips. However, if we do have the data, it can be easily changed to train on this sort of data, by conditioning on the previous style to sample the current style. This probably would reproduce the effect of ElevenLabs, especially for dialogues.

The closest dataset to entire audiobook is LJSpeech, but again it’s completely non-fiction, so it won’t be good for any fiction reading, and it might produce unnatural intonation’s because each clip was treated independently during training.

**Comment by fakerybakery** [comment]
Hmm. Thanks. LibriVox seems like a good place to get public domain audiobooks. Are there any plans to add this capability in the future?

**Comment by yl4579** [comment]
LibriTTS is already taken from LibriVox, but for some reason they aren’t complete audiobook narration but very fragmentized clips taken from complete audiobook narrations. I don’t know why they remove a lot of clips.

**Comment by fakerybakery** [comment]
I feel like the quality would be lower if you trained it on an entire audiobook, right? I don't know, I guess it just feels like the longer the samples are the worse it will be (I might be wrong). Maybe we can use Tortoise TTS's splitting script with this?

**Comment by yl4579** [comment]
No we do have to train on audio clips, but the idea is we condition the current style sampling on previous text and style, so it will be more continuous and possibly also makes it handle dialogue better (if the audio clips are split according to dialogues). It won’t work if we train on entire audio clips because we don’t have enough RAM.

**Comment by fakerybakery** [comment]
Hmm interesting! Are you planning to implement something like this in the future?

**Comment by yl4579** [comment]
Yeah probably, but I don't think it'll be that simple. If the effort is more than trivial concatenation it could be a different project or paper.

**Comment by fakerybakery** [comment]
Hmm. Hypothetically, if there was a long audiobook dataset available, how difficult do you think it would be to implement?

**Comment by fakerybakery** [comment]
I implemented a basic long-text reader on the online demo by splitting text, but it isn't perfect yet. Because I have no limit on how long the audio can be, it has the potential to be abused (if someone puts in an entire book it could slow down the space for everyone else) so I'm currently requiring an "access token" to use it. Since you've joined the Hugging Face org @yl4579, I put the access token in [this private repo](https://huggingface.co/styletts2/access_token_for_beta_features).

However, as you previously mentioned, it will not do well on dialogues as it does not take previous audio into account.

**Comment by MariasStory** [comment]
> I implemented a basic long-text reader on the online demo by splitting text, but it isn't perfect yet. (update: I removed it because someone said it made it harder to clone with Docker)

I am fine with removing the long-text option, because I think that it should be a default setting in every task.
Long text can/should be split and processes.
